src/app/shared/entry/url-deconstruct.pipe.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * Pipe for deconstructing URLs, represented as string objects, into just the important information.
+ *
+ * E.G. https://github.com/agduncan94/hello-dockstore-workflow should be shortened to agduncan94/hello-dockstore-workflow
+ *
+ * @param {string}    providerUrl
+ * @param {string}    versionName
+ *
+ * @return {string}   The pathname of providerUrl
+ */
+
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'urlDeconstruct'
+})
+export class UrlDeconstructPipe implements PipeTransform {
+
+  transform(providerUrl: string, versionName: string): any {
```

Comment:
Put the comment above this.  Many IDEs will automatically generate the jsdoc for it (with the `@param, @return`, etc).  Change return type to string to match your previous comment.